community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/executor/DynamicTaskExecutor.java

```diff
@@ -164,24 +160,35 @@ private void notifyProcessors()
     }
 
     @Override
-    public synchronized void shutdown( int flags )
+    public synchronized void shutdown()
     {
         if ( shutDown )
         {
             return;
         }
 
-        boolean awaitAllCompleted = (flags & TaskExecutor.SF_AWAIT_ALL_COMPLETED) != 0;
-        while ( awaitAllCompleted && !queue.isEmpty() && panic == null /*all bets are off in the event of panic*/ )
+        // Await all tasks to go into processing
+        while ( !queue.isEmpty() && panic == null /*all bets are off in the event of panic*/ )
         {
             parkAWhile();
         }
-        this.shutDown = true;
-        this.abortQueued = (flags & TaskExecutor.SF_ABORT_QUEUED) != 0;
-        while ( awaitAllCompleted && anyAlive() && panic == null /*all bets are off in the event of panic*/ )
+        // Await all processing tasks to be completed
+        for ( Processor processor : processors )
+        {
+            processor.processorShutDown = true;
+        }
+        while ( anyAlive() && panic == null /*all bets are off in the event of panic*/ )
         {
             parkAWhile();
         }
+        this.shutDown = true;
+    }
+
+    @Override
+    public void panic( Throwable panic )
+    {
+        this.panic = panic;
```

Comment:
Multiple calls to `panic` will overwrite each other. That means original cause of failure will be overwritten by subsequent failures.

We could chain them together, although we do not _know_ that they are related. Another idea would be to keep a list of panics and report them all in `assertHealthy`.